File: scimitar-annotations/src/main/java/com/creations/scimitar_annotations/state/ResourceObserver.java
package com.creations.scimitar_annotations.state;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Bind a field with a StateObserver
 * <pre><code>
 * {@literal @}ResourceObserver() StateObserver stateObs;
 * </code></pre>
 */
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
public @interface ResourceObserver {
    String id() default "";
}

File: scimitar-annotations/src/main/java/com/creations/scimitar_annotations/state/ResourceObserver.kt
package com.creations.scimitar_annotations.state

/**
 * Bind a field with a StateObserver
 * <pre>`
 * @ResourceObserver() StateObserver stateObs;
`</pre> *
 */
@Target(AnnotationTarget.FIELD)
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
annotation class ResourceObserver(val id: String = "")
